pkgs/stdenv/generic/make-derivation.nix

```
@@ -441,6 +450,15 @@ else let
                    else true);
     };
 
+    checkedEnv = let
+      envNames = lib.attrNames env;
+      drvNames = lib.attrNames derivationArg;
+    in
+      assert lib.assertMsg (lib.mutuallyExclusive envNames drvNames) "The ‘env’ attribute set cannot contain any attributes passed to derivation. The following attributes are overlapping: ${lib.concatStringsSep ", " (lib.intersectLists envNames drvNames)}";
```

```suggestion
    checkedEnv = 
      assert lib.assertMsg (lib.mutuallyExclusive (lib.attrNames env) (lib.attrNames derivationArg))
        "The ‘env’ attribute set cannot contain any attributes passed to derivation. The following attributes are overlapping: ${lib.concatStringsSep ", " (lib.intersectLists envNames drvNames)}";
```

---

```suggestion
    checkedEnv = let
      overlappingNames = lib.intersectLists (lib.attrNames env) (lib.attrNames derivationArg);
    in
      assert lib.assertMsg (overlappingNames == [])
        "The ‘env’ attribute set cannot contain any attributes passed to derivation. The following attributes are overlapping: ${lib.concatStringsSep ", " overlappingNames}";
```
This avoids the leap that `lib.intersectLists` and `lib.mutuallyExclusive` need to be compatible, and fixes the repetition.

@SuperSandro2000 you missed the second usage of `*Names` in the `lib.intersectLists` calls while inlining.